src/app/OperationalSessionSetup.cpp

```diff
@@ -191,27 +191,29 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     if (mState == State::ResolvingAddress)
     {
         MoveToState(State::HasAddress);
-        err = EstablishConnection();
-        if (err != CHIP_NO_ERROR)
-        {
-            DequeueConnectionCallbacks(err);
-            // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
-            return;
-        }
-    }
-    else
-    {
-        if (!mSecureSession)
+        mInitParams.sessionManager->UpdateAllSessionsPeerAddress(mPeerId, addr);
+        if (!mPerformingAddressUpdate)
```

Comment:
The converse is also true. If they while we're in `ResolvingAddress` with `mPerformingAddressUpdate` being true, if someone calls `FindOrEstablishSession`, they're going to prematurely terminate the address resolution because we're going to attach to an already established session.

Reply:
This I think is harder to solve. It is also less clear this absolutely needs a solution, so I'm fine filling a follow-up.

Reply:
Please look at https://github.com/project-chip/connectedhomeip/pull/21768/files/dd0c55da6d9cb2071119c21597c067cf41d0a3d0#r944666679

For why I don't think this is an issue

Reply:
Missed that, thanks!